### PR TITLE
RealAbs and RealSign

### PR DESCRIPTION
**Pull request by mmatera**
This PR implements two new builtins. Also, its low-level implementation is used to simplify some of the low-level eval functions in arithmetic and testing expressions.

### CODE REVIEW

File: mathics/builtin/arithmetic.py [hunk 1]
```diff
@@ -1237,6 +1281,42 @@ def test(self, expr) -> bool:
...
+class RealSign(Builtin):
+    """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/RealAbs.html</url>
+
+    <dl>
+      <dt>'RealSign[$x$]'
+      <dd>returns $-1$, $0$ or $1$ depending on whether $x$ is negative,
+    zero or positive.
+    </dl>
+    'RealSign' is also known as $sgn$ or $signum$ function.
```
Comment (rocky) [inline mathics/builtin/arithmetic.py:1293]: If it is signum, we should add that before the WMA link a wikipedia link: https://en.wikipedia.org/wiki/Sign_function

File: mathics/eval/testing_expressions.py [hunk 2]
```diff
@@ -6,10 +6,12 @@
 from mathics.core.expression import Expression
 from mathics.core.systemsymbols import SymbolDirectedInfinity
 
-
+# TODO: Remove me. The following function is not used anywhere
```
Comment (rocky) [inline mathics/eval/testing_expressions.py:9]: If it is not used, please remove it.

**Comment by rocky** [comment]
LGTM - just one small comment about documenting RealSign. Merge whenever you feel ready.

**Comment by rocky** [comment]
I see RealSign and RealAbs under [Numerical Functions](https://reference.wolfram.com/language/guide/NumericalFunctions.html)  not [Testing Expressions](https://reference.wolfram.com/language/guide/TestingExpressions.html)

File: mathics/builtin/arithmetic.py [hunk 3]
```diff
@@ -1207,6 +1213,44 @@ class Real_(Builtin):
...
 
+class RealAbs(Builtin):
+    """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/RealAbs.html</url>
+
+    <dl>
+      <dt>'RealAbs[$x$]'
+      <dd>returns the absolute value of a real number $x$.
+    </dl>
+    'RealAbs' is also known as modulus. It is evaluated if $x$ can be compared \
```
Comment (rocky) [inline mathics/builtin/arithmetic.py:1224]: Same comment about Wiki reference.

File: mathics/builtin/testing_expressions/equality_inequality.py [hunk 4]
```diff
@@ -70,7 +70,7 @@ def numerify_args(items, evaluation) -> list:
             for item in items:
                 if not isinstance(item, Number):
                     # TODO: use $MaxExtraPrecision insterad of hard-coded 50
```
Comment (rocky) [inline mathics/builtin/testing_expressions/equality_inequality.py:72]: The TODO is done now. Let's remove the comment.

**Comment by mmatera** [comment]
> I see RealSign and RealAbs under [Numerical Functions](https://reference.wolfram.com/language/guide/NumericalFunctions.html) not [Testing Expressions](https://reference.wolfram.com/language/guide/TestingExpressions.html)

Right now, they are under `mathics.builtin.arithmetic`

**Comment by mmatera** [comment]
Regarding the place, and to reduce the noise, I prefer to keep `RealAbs` and `RealSign` where `Abs`  and `Sign` were before this PR.  Currently, the `mathics.builtin.arithmetic` module is orthogonal to what you can find in `https://reference.wolfram.com/language/guide/ArithmeticFunctions.html`.  At the end of the "arithmetic refactor", we can put the pieces in modules according to the WL documentation.

File: mathics/builtin/arithmetic.py [hunk 5]
```diff
@@ -1283,8 +1290,12 @@ def test(self, expr) -> bool:
 
 class RealSign(Builtin):
     """
-    <url>:WMA link:https://reference.wolfram.com/language/ref/RealAbs.html</url>
-
+    <url>
+        :Signum:
```
Comment (rocky) [inline mathics/builtin/arithmetic.py:1294]: The link name should be what Wikipedia calls it, so this should be "Sign function". Wikipedia mentions that this is also called "Signum"

Comment (mmatera) [reply mathics/builtin/arithmetic.py:1294]: Ok, I can change this in another iteration.

Comment (mmatera) [reply mathics/builtin/arithmetic.py:1294]: I have included that change in #886, which is the next round of patches.

**Pull request merged**